polishcalc.cc

```diff
@@ -0,0 +1,31 @@
+#include <assert.h>
+#include <memory>
+#include "polishcalc.h"
+#include "polishcalcutils.h"
+#include "operations.h"
+
+namespace PolishCalcComponent {
+
+value_t PolishCalc::calculate(const std::string &inputStr)
+{
+    std::vector<std::string> tokens;
+    PolishCalcComponent::split2Tokens(inputStr, tokens);
+
+    std::unique_ptr<IOperandsMemory> memory(makeMemory());
+
+    processTokens(tokens, memory.get());
```

Comment:
if `processTokens` does not control  memory life-cycle I would provide it by reference